handlers/handlers.go

```diff
@@ -246,8 +249,51 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content  string `json:"content"`
+			Category string `json:"category"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := models.CreateNewNote(userId, noteForm.Content)
+
+		if err := noteservice.StoreNewNote(note); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if !(noteForm.Category == "") {
+
+			category, err := models.DeserializeCategory(strings.ToLower(noteForm.Category))
```

Comment:
we shouldn't have to use `ToLower`

Reply:
This is the only time we sanitize the string. Since the input is from the outside world this is the correct place to sanitize.

Reply:
Yeah it's from the outside world, but we set the rules of our api, not the outside world. This isn't "sanitizing", it's widening the amount of inputs we accept. The outside world should match exactly the api we set forth, otherwise they get a 4xx. I expect this won't be a problem, since we've standardized on these strings and made them lowercase all over the codebase